packages/hdwallet-web3-portis/src/adapter.ts

```diff
@@ -0,0 +1,36 @@
+import {
+    Keyring,
+    HDWallet
+  } from '@shapeshiftoss/hdwallet-core'
+
+import { PortisTransport } from './PortisTransport'
+import { Web3PortisHDWallet } from './web3Portis'
+
+  export class Web3PortisAdapter {
+    keyring: Keyring
+  
+    private constructor (keyring: Keyring) {
+      console.log('portis adapter constructor')
+      this.keyring = keyring
+    }
+  
+    public static useKeyring (keyring: Keyring) {
+      console.log('portis adapter useKeyring')
+      return new Web3PortisAdapter(keyring)
+    }
+  
+    public async initialize (portis: any): Promise<number> {
+      console.log('portis adapter initialize')
+      let transport = new PortisTransport(this.keyring)
+      let wallet = new Web3PortisHDWallet(portis, transport)
+      await wallet.initialize()
+      this.keyring.add(wallet)
+      return Object.keys(this.keyring.wallets).length
+    }
+
+    public async pairDevice (deviceId: string): Promise<HDWallet> {
+      console.log('portis adapter pairDevices', deviceId)
+      const wallet = this.keyring.get(deviceId)
```

Comment:
What did we decide `deviceId` meant for Portis wallets?

Reply:
@keepkeyjon as of now, wallets are identified by their corresponding eth address, but idk if that's the best long term solution. The ETH address for the wallet is what gets passed in every time the `portis.onActiveWalletChange` callback gets invoked.